napari/utils/perf/_trace_file.py

```diff
@@ -71,22 +84,33 @@ def write_event(self, event: PerfEvent) -> None:
         event : PerfEvent
             Event to write.
         """
+        category = "none" if event.category is None else event.category
+
         # Event type "X" denotes a completed event. Meaning we already
         # know the duration. The format wants times in micro-seconds.
         data = {
-            "pid": self.pid,
+            "pid": event.pid,
+            "tid": event.tid,
             "name": event.name,
-            "cat": event.category,
-            "ph": "X",
+            "cat": category,
+            "ph": event.type,
             "ts": event.start_us,
-            "dur": event.duration_us,
+            "args": event.args,
         }
+
+        if event.type == "X":
+            data["dur"] = event.duration_us
+        else:
+            assert event.type == "I"
+            # For instant events the scope "s" is one of:
+            #     "g" - global
+            #     "p" - process
+            #     "t" - thread
+            data["s"] = "p"
```

Comment:
It's unclear to me what scope means in this context or why it would always be p here?

Reply:
The "scope" here is strictly a chrome://tracing term. It's saying what type of "instant event" is this? Is it an event that applies only to one thread, to one process, or to the whole system. It basically just comes down to when `chrome://tracing` draws the "instant event" where does it draw on the line.

I just hardcoded it to process based on how I was using it. Probably we should expose all 3 options since it's not hard. I just didn't even experiment with them to see what they looked like, so I didn't want to expose them without at least trying them.

Reply:
I added a better comment.